metadata/src/main/java/org/apache/kafka/controller/FeatureControlManager.java

```diff
@@ -266,6 +311,11 @@ private ApiError updateMetadataVersion(
             new FeatureLevelRecord()
                 .setName(MetadataVersion.FEATURE_NAME)
                 .setFeatureLevel(newVersionLevel), FEATURE_LEVEL_RECORD.lowestSupportedVersion()));
+
+        // If we are moving to a version that supports migrations, we need to write the correct state into the log
```

Comment:
I don't see why this is needed. If we're coming from IBP_3_3_IV3 or earlier, the only valid value is NONE. And omitting the record just means that its value is implictly assumed to be equal to NONE. So there's nothing to do.

Reply:
Actually, that's not quite true... we should return an error if someone is trying to change metadataversion and we're in a state other than NONE or POST_MIGRATION. Since changing MV during a migration is a big headache and we agreed to disable that.